### PR TITLE
Refactor Error-Handling

### PR DESCRIPTION
**Pull request by radlinskii**
might be easier to review it looking through commits one-by-one

### CODE REVIEW

**Comment by radlinskii** [comment]
You can push some more changes to this branch if it's related to error handling. There is no need to merge it straight away.

File: parser/parser.go [hunk 1]
```diff
@@ -233,7 +240,6 @@ func (p *Parser) parseExpressionStatement() *ast.ExpressionStatement {
 	stmnt := &ast.ExpressionStatement{Token: p.curToken}
 
 	stmnt.Expression = p.parseExpression(LOWEST)
-	p.nextToken()
```
Comment (radlinskii) [inline parser/parser.go]: This was the bug that made some illegal characters never to be found.

File: parser/parser.go [hunk 2]
```diff
@@ -200,7 +203,18 @@ func (p *Parser) expectPeek(t token.Type) bool {
...
+}
+
+func (p *Parser) checkIfOverridesBuiltin() {
+	if _, ok := builtins[p.curToken.Literal]; ok {
+		msg := fmt.Sprintf("cannot override built-in function: %q at line: %d", p.curToken.Literal, p.curToken.LineNumber)
+		p.errors = append(p.errors, msg)
+	}
+}
 
+func (p *Parser) semiError() {
```
Comment (agnieszka-miszkurka) [inline parser/parser.go]: semi means half xd maybe missingSemicolonError or sth similar would be better?

Comment (radlinskii) [reply parser/parser.go]: well, go ahead then 😄

**Pull request merged**